### ISSUE TITLE
Add anyObject to MockitoMatchers

### ISSUE THREAD

**Issue by nvartolomei** [etorreborre/specs2#578]

**Comment by nvartolomei** [comment]
Actually I think, if no type tag is passed to any then it should under the hood call org.mockito.Matchers.any without any argument, so that it matches everything and not `Nothing` type.

**Comment by etorreborre** [comment]
It seems to be working indeed like this:
```scala
  def any[T]: T = org.mockito.Matchers.any.asInstanceOf[T]
```

What kind of issue did you have with the previous version?

**Comment by nvartolomei** [comment]
@etorreborre can you please rephrase your question.

**Comment by etorreborre** [comment]
Sure I was writing down another version of `any`, which is not requiring a `ClassTag`, and which seemed to be passing all my tests. And I was wondering what kind of issue the current version of specs2 was creating for you because your description doesn't really say. You can also hop on [gitter](https://gitter.im/etorreborre/specs2) if you want

**Comment by nvartolomei** [comment]
Using specs2 with Mockito 1.*, type passed to any is ignored by Mockito's `any` matcher as noted in [code comment](https://github.com/mockito/mockito/blob/release/1.x/src/org/mockito/Matchers.java#L290).

When settings up mocks, you can add a set expectations in 2 ways and both of them have the same effect:
```scala
trait A
class B extends A
class C extends A
// 1
myObject.method(any, any) returns 1
// 2
myObject.method(any[B], any[Boolean]) returns 1
```

The problem is that if when I verify invocations, all checks bellow will pass, because Mockito matchers does not check types.

```scala
there was one(myObject).method(any, any)
there was one(myObject).method(any[B], any[Boolean])
there was one(myObject).method(any[C], any[Boolean]) // <- oops
```

Mockito 2.* added type checks as noted in [code comment](https://github.com/mockito/mockito/blob/release/2.x/src/main/java/org/mockito/ArgumentMatchers.java#L206).

Now I was expecting two of the checks above to be pass, except last one.

```scala
there was one(myObject).method(any, any)
there was one(myObject).method(any[B], any[Boolean])
```

But beside this, first check does not pass too.

The change you proposed above will keep the old behaviour when using Mockito 2, instead I want the new behaviour, but also to be able to use old behaviour when using `any` without an explicit TypeTag.

To sum up:
`specs any should use mockito's any()`
`specs any[T] should use mockito's any(Class<T> type)`

**Comment by etorreborre** [comment]
I upgraded specs2 to Mockito 2.x with:
```
def any[T : ClassTag]: T = 
  org.mockito.ArgumentMatchers.any(implicitly[ClassTag[T]].runtimeClass).asInstanceOf[T]
def any[T]() = 
  org.mockito.ArgumentMatchers.any.asInstanceOf[T]
// alias for any()
def anyObject[T] = 
  org.mockito.ArgumentMatchers.any.asInstanceOf[T]
```
We can not overload directly `any` without parentheses otherwise calls are ambiguous.
I will publish an interim version for you to test shortly.

**Comment by etorreborre** [comment]
Can you please try `3.8.9-20170519080524-c1457d5`?

**Comment by etorreborre** [comment]
I merged those changes into the master. Please reopen if something is not working.
